core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathPropertiesImpl.kt

```
@@ -63,7 +66,20 @@ data class PathPropertiesImpl(val infra: TrackProperties, val chunkPath: ChunkPa
     }
 
     override fun getSpeedLimits(trainTag: String?): DistanceRangeMap<Speed> {
-        return getRangeMap { dirChunkId -> infra.getTrackChunkSpeedSections(dirChunkId, trainTag) }
+        return getRangeMap { dirChunkId ->
+            val route =
+                pathRoutes?.let { routes ->
```

I'd prefer if we assert that the routes have been defined here, and they shouldn't be optional in functions called from here.

The parameter may be optional when instantiating the object because we sometimes use them without accessing the speed limits. But when we actually do get the value, the result would be invalid if they're unspecified. 

---

From what I see, the routes aren't set in STDCM even though we run simulations. We'd need to change `infraExplorer.getCurrentEdgePathProperties`. 

There are several ways to get the route in this context, I think the most straightforward would be to build a block -> route mapping in the `infraExplorer` object.

---

Done / done (maybe?...)

---

Nice job on specifying the routes in stdcm context

The empty list as a default value was not what I had in mind, sorry about that.
What I'd want to avoid would be to fail silently when getting speed limits on a path that haven't specified the routes. It's likely to happen *at some point in the future*, the results would be invalid, and it would be near impossible to figure it out.

IMO the route list should be optional in `PathPropertiesImpl`. When getting seed limits (and only then), an error should be thrown if it's still null at this point.

---

Ok, I've added the assert and set the routes to an empty list explicitly where needed. While doing so I spotted a bug in my STDCM changes with the (block -> pathProp) cache, so you might want to check that part again. 

I left a bunch of FIXMEs in places where I'm not sure if an empty route list is fine or not (`EngineeringAllowanceManager.kt` and `STDCMPostProcessing.kt`).

---

What I meant was to never use an empty list as a default value.

We keep the field nullable in the class. When accessing max speed, we raise an error if it's null (the result would be wrong). From this point only we can transition to non-optional types.

That would remove any ambiguity on whether we need to set the route list: if it crashes, it's required. I do believe it's required in engineering allowances and in stdcm post-processing.

---

Should be ok now.